### PR TITLE
Fixed unwanted dynamic allocations <master> [9831]

### PR DESCRIPTION
**Pull request by MiguelCompany**

### CODE REVIEW

File: src/cpp/rtps/participant/RTPSParticipantImpl.cpp [hunk 1]
```diff
@@ -622,7 +622,11 @@ bool RTPSParticipantImpl::create_writer(
 
     std::lock_guard<std::recursive_mutex> guard(*mp_mutex);
     m_allWriterList.push_back(SWriter);
-    if (!is_builtin)
+    if (is_builtin)
+    {
+        async_thread().wake_up(SWriter);
+    }
+    else
```
Comment (IkerLuengo) [inline src/cpp/rtps/participant/RTPSParticipantImpl.cpp:629]: What about async user writers? We could be checking 
`param.mode == ASYNCHRONOUS_WRITER`

Comment (MiguelCompany) [reply src/cpp/rtps/participant/RTPSParticipantImpl.cpp:629]: It doesn't matter. The goal here is to create the async thread as soon as possible. As the PDP announcer will be always created when creating the participant, the first created writer will always be a builtin one.

**Review by IkerLuengo — APPROVED** [review]
LGTM

**Review by IkerLuengo — APPROVED** [review]
:+1:

**Comment by richiware** [comment]
Build status:
* Linux [Build Status](http://jenkins.eprosima.com:8080/job/FastRTPS%20Manual%20Linux/3099/)
* Linux aarch64[Build Status](http://jenkins.eprosima.com:8080/job/FastRTPS%20Manual%20Linux%20aarch64/318/)
* Mac [Build Status](http://jenkins.eprosima.com:8080/job/FastRTPS%20Manual%20Mac/2772/)
* Windows [Build Status](http://jenkins.eprosima.com:8080/job/FastRTPS%20Manual%20Windows/3130/)

**Pull request merged**